### PR TITLE
snyk security vuln. - update sshpk version

### PR DESCRIPTION
**Pull request by knoxcard**
✗ High severity vulnerability found on sshpk@1.13.1
- desc: Regular Expression Denial of Service (ReDoS)
- info: https://snyk.io/vuln/npm:sshpk:20180409
- from: node_services@1.0.0 > node-gyp@3.6.2 > request@2.83.0 > http-signature@1.2.0 > sshpk@1.13.1
Your dependencies are out of date, otherwise you would be using a newer sshpk than sshpk@1.13.1.

### CODE REVIEW

**Comment by lc3t35** [comment]
+1

**Comment by austinkelleher** [comment]
Can we please get this merged @arekinath @davidlehn? Thanks in advance.

**Comment by davidlehn** [comment]
@austinkelleher I'm not a maintainer here, just a former contributor and user.  Note that in this case it's likely a semver based install will use the newer sshpk version anyway.

**Comment by wyardley** [comment]
FWIW, even if I remove `node_modules` and reinstall, I get an nsp warning for https://nodesecurity.io/advisories/606, e.g.,
`opsgenie-sdk@0.4.2 > requestretry@1.13.0 > request@2.83.0 > http-signature@1.2.0 > sshpk@1.13.1`
or
` @google-cloud/pubsub@0.18.0 > google-gax@0.16.1 > grpc@1.8.4 >  node-pre-gyp@0.6.39 > request@2.83.0 > http-signature@1.2.0 > sshpk@1.13.1`

**Comment by NejcZdovc** [comment]
@wyardley security problem was fixed with `1.14.1`, where you have version 1.13.1 installed

**Comment by knoxcard** [comment]
GitHub needs a community pull voting system that does not require permission of a maintainer.  This project has 24 contributors, but appears to only have one maintainer @arekinath?  Looks like I need to submit a pull request for GitHub to allow for this? :-)

**Comment by benwiggins** [comment]
Or you could simply not create unnecessary pull requests.

You're pulling in an old, out of date dependency, most likely caused by your own package lock file.

**Comment by knoxcard** [comment]
@benwiggins - haha, I was unaware of this, thanks for educating me about this. Right on.

**Comment by knoxcard** [comment]
On that note, closing ticket and exit building...

**Comment by davidlehn** [comment]
Lock file nonsense aside, this PR is still a good idea to ensure a proper sshpk version is used.

**Comment by knoxcard** [comment]
So here is what I am g

**Comment by benwiggins** [comment]
Don’t get me wrong, I 100% agree that bumping the minimum version to a known “good” version is good practice / courtesy.

But it does nothing to educate those tearing the hair out creating pull requests and issues because they don’t understand how dependency resolution/semvers/package locks actually work. 

Nor does it necessarily solve someone’s nsp or snyk “security issue” if your package is a nested(-nested-nested) dependency, which in this case being a request dependency, is pretty likely. It just moves the pull request/issue noise somewhere else :)

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   "dependencies": {
     "assert-plus": "^1.0.0",
     "jsprim": "^1.2.2",
-    "sshpk": "^1.7.0"
+    "sshpk": "^1.14.1"
```
Comment (BYK) [inline package.json]: This should now be `^1.14.2` to account for the `new Buffer` usage fix. See https://github.com/joyent/node-sshpk/pull/46

Comment (knoxcard) [reply package.json]: Done, resubmitted pull request with 1.14.2

Comment (knoxcard) [reply package.json]: What's the hold up?

Comment (ignaciobOktana) [reply package.json]: Sorry for that @BYK
@knoxcard i'm working on fixing snyk vulnerabilities generated by `sshpk` version 1.7 that seems to be fixed on newest version 1.14.2. 
I was wondering if there is a release date with that package updated.
Thanks!!

**Comment by kusor** [comment]
Done as of PR #86

**Pull request closed without merging**